<launch>
  <!-- Launch Real Sense Node -->
  <!-- <include file="$(find realsense2_camera)/launch/rs_camera.launch"/> -->

  <!-- Camera topic (Real Sense) -->
  <!-- <arg name="camera_topic" default="/camera/color/image_raw" />
  <arg name="img_compressed" default="false" /> -->

  <!-- <arg name="camera_topic" default="/camera/color/image_raw/compressed" /> -->
   <!-- <arg name="img_compressed" default="true" /> -->

  <!-- <arg name="camera_topic" default="/xtion/rgb/image_raw/compressed" /> -->
  <!-- <arg name="img_compressed" default="true" /> -->

  <arg name="camera_topic" default="/object_detector/detection_image/compressed" />
  <arg name="img_compressed" default="true" />


  <!-- Visualization -->
  <arg name="visualization" default="true" />

  <!-- MediaPipe Holistic Illustration Variables -->
  <arg name="drawPoseLandmarks" default="true" />
  <arg name="drawFaceLandmarks" default="false" />
  <arg name="drawRightHandLandmarks" default="true" />
  <arg name="drawLeftHandLandmarks" default="true" />
  <arg name="drawFaceBoundary" default="false" />

  <!-- Use Hands to Determine Pointing Direction (true). Otherwise use Arms to determine pointing direction (false) -->
  <arg name="pointing_hands" default="false" />

  <!-- Definition of Pointing Hand/Arm Msg -->
  <arg name="pointing_right_hand_msg" default="right" />
  <arg name="pointing_left_hand_msg" default="left" />

  <!-- Launch Mediapipe Holistic Node -->
  <node ns="perception" name="mediapipe_holistic" pkg="perception_tests" type="mediapipeHolisticnode.py" output="screen">
    <param name="camera_topic" value="$(arg camera_topic)" type="string"/>
    <param name="img_compressed" value="$(arg img_compressed)" type="bool"/>
    <param name="pointing_hands" value="$(arg pointing_hands)" type="bool"/>
    <param name="visualization" value="$(arg visualization)" type="bool"/>
    <param name="drawPoseLandmarks" value="$(arg drawPoseLandmarks)" type="bool"/>
    <param name="drawFaceLandmarks" value="$(arg drawFaceLandmarks)" type="bool"/>
    <param name="drawRightHandLandmarks" value="$(arg drawRightHandLandmarks)" type="bool"/>
    <param name="drawLeftHandLandmarks" value="$(arg drawLeftHandLandmarks)" type="bool"/>
    <param name="drawFaceBoundary" value="$(arg drawFaceBoundary)" type="bool"/>
    <param name="pointing_right_hand_msg" value="$(arg pointing_right_hand_msg)" type="string"/>
    <param name="pointing_left_hand_msg" value="$(arg pointing_left_hand_msg)" type="string"/>
  </node>


</launch>
